include/roboteam_ai/world_new/World.hpp

```diff
@@ -0,0 +1,178 @@
+//
+// Created by john on 12/16/19.
+//
+
+#ifndef RTT_WORLD_HPP
+#define RTT_WORLD_HPP
+#include <vector>
+#include "views/WorldDataView.hpp"
+#include "roboteam_proto/RobotFeedback.pb.h"
+#include "WorldData.hpp"
+
+namespace rtt::world_new {
+
+    /**
+     * Structure that represents the world and history of the world.
+     *
+     * This structure operates under the assumption of an immutable state
+     * Adjusting any value inside this world after construction may
+     * result in undefined behavior. Any sort of mutation of already constructed
+     * WorldData will break the entire system and introduce dataraces due to
+     * a shared immutable state.
+     *
+     * WorldData is a snapshot of the current world as seen by the camera
+     * Feeback is already applied to the robots when they are constructed
+     * As soon as a new world is constructed, which should only be done
+     * through updateWorld, the current world is pushed to history, given that
+     * this is a Some variant. Using any other unconventional way to push
+     * a world to history will result in undefined behavior.
+     *
+     * The entire system with World relies on immutability,
+     * this is something that Rust and haskell enforce and is seen as
+     * one of the only ways to keep a project datarace-free
+     */
+    class World {
+    public:
+        /**
+         * Global singleton for World, scott-meyers style
+         * @return A pointer to a static World
+         */
+        inline static World* instance() {
+            static World worldInstance{ &rtt::SETTINGS };
```

Comment:
You still sure you want a singleton? Or is it temporary? I'd prefer parameter passing

Reply:
please don't go changing that now by the way

Reply:
Meant to be temporary, better than a full on global i guess